src/components/Interpretations/InterpretationsProvider/InterpretationsManager.js

```diff
@@ -0,0 +1,307 @@
+        this.currentVisualizationId = id
+        const interpretations = await this.fetchInterpretationsList()
+        this.resetInterpretations(interpretations)
+        return groupInterpretationIdsByDate(interpretations)
+    }
+
+    async createInterpretation({ type, id, text, onComplete }) {
+        await this.mutate(
+            {
+                resource: `interpretations/${type}/${id}`,
+                type: 'create',
+                data: text,
+            },
+            { onComplete }
+        )
+        /* since the create request does not rerun an ID we must refetch the list
+         * and cannot return the created interpretation */
+        const interpretations = await this.fetchInterpretationsList()
+        this.resetInterpretations(interpretations)
+        this.notifyInterpretationsListObserver()
+        return null
+    }
+
+    async deleteInterpretation({ id, onComplete, onError }) {
+        await this.mutate(
+            {
+                resource: 'interpretations',
+                id,
+                type: 'delete',
+            },
+            { onComplete, onError }
+        )
+        // This happens when deleting the interpretation from the modal
+        if (this.activeInterpretationId && id === this.activeInterpretationId) {
+            this.clearActiveInterpretation()
+        }
+        this.interpretations.delete(id)
+        this.notifyInterpretationsListObserver()
+        return null
+    }
+
+    async updateInterpretationText({ id, text, onComplete, onError }) {
+        await this.mutate(
+            {
+                resource: 'interpretations',
+                type: 'update',
+                partial: false,
+                id,
+                data: text,
+            },
+            { onComplete, onError }
+        )
+        const updatedInterpretation = {
+            ...this.getInterpretation(id),
+            text,
+        }
+        this.interpretations.set(id, updatedInterpretation)
+        this.notifyInterpretationObservers(id)
+        return updatedInterpretation
+    }
+
+    async toggleInterpretationLike(id) {
+        const interpretation = this.getInterpretation(id)
+        const wasLikedByCurrentUser = interpretation.likedBy.some(
+            (likedBy) => likedBy.id === this.currentUser.id
+        )
+        await this.mutate({
+            resource: `interpretations/${id}/like`,
+            type: wasLikedByCurrentUser ? 'delete' : 'create',
+        })
+        const isLikedByCurrentUser = !wasLikedByCurrentUser
+        const updatedInterpretation = {
+            ...interpretation,
+            likedBy: isLikedByCurrentUser
+                ? interpretation.likedBy.concat({
+                      id: this.currentUser.id,
+                  })
+                : interpretation.likedBy.filter(
+                      (lb) => lb.id !== this.currentUser.id
+                  ),
+            likes: isLikedByCurrentUser
+                ? interpretation.likes + 1
```

Comment:
I understand why we are doing this and I suppose we were doing something similar before.
The potential issue is that you get an out of sync counter, but I guess it's not that important, it will be fixed whenever the interpretations panel is re-opened.